wrapanapi/systems/openstack.py

```diff
@@ -634,9 +634,23 @@ def tenant_api(self):
 
     @property
     def capi(self):
-        if not self._capi:
-            self._capi = cinderclient.Client(session=self.session, service_type="volume")
-        return self._capi
+        if self._capi:
+            return self._capi
+
+        # Detect the higest possible hopefully working cinder version
+        service_types = 'volumev3 volumev2 volume'.split()
+        for service_type in service_types:
+            try:
+                capi = cinderclient.Client(session=self.session, service_type=service_type)
```

Comment:
Do we even need to pass the service_type here? It appears to work without passing one at all, at least against rhos13.

Reply:
Well IIRC `list` used to be working, but the important was to make this work:
```
     volume = provider.mgmt.create_volume(1, imageRef=provider.mgmt.get_template(image).uuid)
```

It couldn't find the image in glance( specified by the imageRef) if `service_type=volume` was used. 

As it seems without specifying the service type, the `create_template` from image works, so we need to iterate between the versions to choose the working one and rather rely on the cinder client to choose the best one.